RecoParticleFlow/PFClusterProducer/plugins/Basic2DClusterForEachSeed.cc

```diff
@@ -0,0 +1,52 @@
+#include "Basic2DClusterForEachSeed.h"
+
+void Basic2DClusterForEachSeed::buildClusters(const edm::Handle<reco::PFRecHitCollection>& input,
+                                              const std::vector<bool>& rechitMask,
+                                              const std::vector<bool>& seedable,
+                                              reco::PFClusterCollection& output) {
+  auto const& hits = *input;
+  std::vector<bool> used(hits.size(), false);
+  std::vector<unsigned int> seeds;
+
+  // get seeds
+  seeds.reserve(hits.size());
+  for (unsigned int i = 0; i < hits.size(); ++i) {
+    if (!rechitMask[i] || !seedable[i])
+      continue;
+    seeds.emplace_back(i);
+  }
+
+  // loop over seeds and make clusters
+  reco::PFCluster cluster;
+  for (auto seed : seeds) {
```

Comment:
do you mean something like this instead ?
```suggestion
  for (unsigned int seed = 0; seed < hits.size(); ++i) {
```

Reply:
Exactly

Reply:
Well, without removing
```C++
  // loop over seeds and make clusters
  reco::PFCluster cluster;
```

Reply:
Added back :-)

Reply:
Thank you @perrotta @fwyzard .
I think I incorporated your suggestions and used this opportunity to further simplify it.